r/R/dplyr-funcs-datetime.R

```diff
@@ -826,3 +827,54 @@ register_bindings_datetime_rounding <- function() {
     }
   )
 }
+
+register_bindings_hms <- function() {
+  numeric_to_time32 <- function(x) {
+    # The only numeric which can be cast to time32 is int32 so double cast to make sure
+    cast(cast(x, int32()), time32(unit = "s"))
+  }
+
+  register_binding(
+    "hms::hms",
+    function(seconds = 0, minutes = 0, hours = 0, days = 0) {
+      total_secs <- seconds +
+        Expression$create("multiply_checked", minutes, 60) +
+        Expression$create("multiply_checked", hours, 3600) +
+        Expression$create("multiply_checked", days, 86400)
+
+      return(numeric_to_time32(total_secs))
+    }
+  )
+
+  register_binding(
+    "hms::as_hms",
+    function(x = numeric()) {
+      datetime_to_time32 <- function(datetime) {
+        hour <- call_binding("hour", datetime)
+        min <- call_binding("minute", datetime)
+        sec <- call_binding("second", datetime)
+
+        return(call_binding("hms::hms", seconds = sec, minutes = min, hours = hour))
+      }
```

Comment:
Any reason not to put this up on the same level as `numeric_to_time32()`?

Reply:
Nope, updated now!